pkg/objectio/reader.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objectio
+
+import (
+	"context"
+	"github.com/matrixorigin/matrixone/pkg/fileservice"
+)
+
+type ObjectReader struct {
+	object *Object
+	name   string
+}
+
+func NewObjectReader(name string, fs fileservice.FileService) (Reader, error) {
+	reader := &ObjectReader{
+		name:   name,
+		object: NewObject(name, fs),
+	}
+	return reader, nil
+}
+
```

Comment:
ReadMeta -- do we really want to read one by one?    What if we need to scan a file of a million blocks?   Are we going to do one million IO?